docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -0,0 +1,1946 @@
+automatically improve contracts’ functionality without requiring any
+error prone migrations, without downtime, without requiring any
+additional network traffic, and without any extensive communication with
+downstream users.
+
+Requirements
+------------
+
+Note that SCU is only available when the criteria below is met:
+
+-  Canton 2.9.x or above
+
+-  Daml LF Version 1.16 or above
+
+-  Canton Protocol Version 6 or above
+
+There are instructions below on how to configure this setup. The
+sections below, unless explicitly stated otherwise, assume that this is
+the case.
+
+How Does it Work
+----------------
+
+When upgrading a package, the package author modifies their existing
+package to add new functionality, such as new fields and choices. When
+the new package is uploaded to a participant with the old version, the
+participant ensures that it can automatically upgrade every change
+between the old and new version before adding it to the package set.
+
+In order to be able to automatically upgrade a contract or datatype, SCU
+restricts the kinds of changes that a new package version can introduce
+over its prior version.
+
+For example, the simplest kind of data transformation that SCU supports
+is adding a field to a template.
+
+Given the following first version of a template:
+
+.. code:: daml
+
+  template IOU
+    with
+      issuer: Party
+      owner: Party
+      value: Int
+    where
+      signatory issuer
+      observer owner
+
+We can add a new field for currency:
+
+.. code:: daml
+
+  template IOU
+    with
+      issuer: Party
+      owner: Party
+      value: Int
+      -- New field:
+      currency: Optional String
+    where
+      signatory issuer
+      observer owner
+
+With SCU, any new template fields must be optional - templates from the
+old version are automatically upgraded to new versions by setting the
+new field to ``None``. This optional field requirement extends to all
+records in your package. Conversely, newer contracts with this field set
+to ``None`` can be automatically downgraded to previous versions of the
+template in workflows that have not yet been updated.
+
+Automatic Data Upgrades
+~~~~~~~~~~~~~~~~~~~~~~~
+
+When extending data in a daml model, SCU requires the old model be
+representable in the new model. I.e., for extending a record, we can
+only add nullable (``Optional``) fields, so that old data can be represented
+by setting these fields to ``None``. This mentality is continued for
+Variants and Enums, which only allow adding constructors, with some
+other restrictions covered in `Continuing to Write Your Upgrades <#continuing-to-write-your-upgrades>`__. This
+approach is inspired by
+`Protobuf <https://protobuf.dev/programming-guides/proto3/#updating>`__
+and Typescript's ability to ignore `excess
+fields <https://www.typescriptlang.org/docs/handbook/2/objects.html#excess-property-checks>`__
+via ``as``.
+
+Following is a list of places where automatic data upgrades occur:
```

Comment:
Below you also refer to downgrades, but they are not explained at this point. Consider explaining them together with the upgrades

Reply:
It is already briefly explained at the end of the above "How does it work" section, do you think we need more than that?

Reply:
so are upgrades ;-) 

So why not talk about "Automatic Data Upgrades and Downgrades" and treat them equally?